src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,343 @@
+)
+    @eval begin
+        function $(func)(iv::InterfaceValues, qp::Int, i::Int)
+            nbf = getnbasefunctions(iv)
+            nbf_a = getnbasefunctions(iv.face_values_a)
+            if i <= nbf_a
+                fv = iv.face_values_a
+                f_value = $(f_)(fv, qp, i)
+                return $(multiplier) * f_value
+            elseif i <= nbf
+                fv = iv.face_values_b
+                f_value = $(f_)(fv, qp, i - nbf_a)
+                return $(multiplier) * f_value
+            end
+            error("Invalid base function $i. Interface has only $(nbf) base functions")
+        end
+    end
+end
+
+for (func,                      f_,                 ) in (
+    (:shape_value_jump,         :shape_value,       ),
+    (:shape_gradient_jump,      :shape_gradient,    ),
+    (:geometric_value_jump,     :geometric_value,   ),
+)
+    @eval begin
+        function $(func)(iv::InterfaceValues, qp::Int, i::Int, normal_dotted::Bool = true)
+            f_value = $(f_)(iv, qp, i)
+            nbf_a = getnbasefunctions(iv.face_values_a)
+            if i <= nbf_a
+                normal_dotted || return f_value
+                multiplier = getnormal(iv, qp, true)
+                return f_value isa Number ? f_value * multiplier : f_value ⋅ multiplier
+            else
+                normal_dotted || return -f_value
+                multiplier = getnormal(iv, qp, false)
+                return f_value isa Number ? f_value * multiplier : f_value ⋅ multiplier
+            end
+        end
+    end
+end
+
+"""
+    function_value_average(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u))
+
+Compute the average of the function value at the quadrature point on interface.
+"""
+function_value_average
+
+"""
+    function_value_jump(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u), normal_dotted::Bool = true)
+
+Compute the jump of the function value at the quadrature point over the interface.
+
+`normal_dotted::Bool` determines whether to use the definition ``\\llbracket v \\rrbracket=v^- -v^+`` if it's `false`, or
+ the definition  ``\\llbracket v \\rrbracket=v^- ⋅ \\vec{n}^- + v^+ ⋅ \\vec{n}^+`` if it's `true`, which is the default.
+
+!!! note
+    If `normal_dotted == true` then the jump of scalar function values is a vector.
+"""
+function_value_jump
+
+"""
+    function_gradient_average(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u))
+
+Compute the average of the function gradient at the quadrature point on the interface.
+"""
+function_gradient_average
+
+"""
+    function_gradient_jump(iv::InterfaceValues, qp::Int, u::AbstractVector, dof_range = eachindex(u), normal_dotted::Bool = true)
+
+Compute the jump of the function gradient at the quadrature point over the interface.
+
+`normal_dotted::Bool` determines whether to use the definition ``\\llbracket \\vec{v} \\rrbracket=\\vec{v}^- -\\vec{v}^+`` if it's `false`, or
```

Comment:
I think we should settle on one single definition for the jump, document it, and show how one can compute the other version manually using `function_value_here` and `function_value_there`.

Reply:
Without any experience on the subject (So take this with a bucket of salt): 
Would a general definition be to use the dyadic product for the jump, $[f] = f_a \otimes n_a + f_b \otimes n_b$, instead of the dot product, `here-there`, or `there-here`?
```
function foo_jump(iv, q_point, xe_a, xe_b, range_a, range_b)
    fa = foo(iv.a, q_point, xe_a, range_a)
    na = getnormal(iv.a, q_point)
    fb = foo(iv.b, q_point, xe_b, range_b)
    nb = getnormal(iv.b, q_point) # Or nb = -na
    return fa ⊗ na + fb ⊗ nb
end
```
Vector functions: This retains the tangential jump for vector functions (as opposed to the dot definition in the [ref](https://www.researchgate.net/publication/211746421_Unified_Analysis_of_Discontinuous_Galerkin_Methods_for_Elliptic_Problems)). The signed normal jump is `- n ⋅ foo_jump() ⋅ n` where `n` is `na` or `nb` (`na=-nb` following the same arguments as for the quadrature weights)

Scalar functions: Vector jump as in [ref](https://www.researchgate.net/publication/211746421_Unified_Analysis_of_Discontinuous_Galerkin_Methods_for_Elliptic_Problems) is `(foo_jump() ⋅ n)*n`. But here I guess often the absolute value of the jump suffices, in which case the dyadic implementation would be inefficient (but it would be a 1-liner for users to do that efficiently as Fredrik writes).

Reply:
But continuing thinking out loud, I guess having the jump as, `ub-ua`, and `getnormal(iv, q_point)` giving one normal (probably `na` in this case) would be as general and more performant!

Reply:
Not so sure. Where does the assumption come from that the normals have to align for all points in time? I think this breaks down on moving domains.

Reply:
In the Lagrangian description, I think that should be valid since the geometric nodes are matching for `InterfaceValues` AFAIU. 

For Eulerian descriptions, even the same `getdetJdV` is not true (since the faces can change area due to the deformation). And currently, I think an Eulerian description is difficult in Ferrite, for example, the derivatives of, e.g., `dV` and shape functions, wrt. cell coordinates are not available*. Furthermore, in the case that normals don't align and the jump is calculated as `ua*na + ub*nb`, if `ua=ub`, but `na!=-nb`, the jump would be nonzero which doesn't make sense to me. Hence in any case a single normal should be used IMO, where one could argue that e.g. `(na-nb)/|na-nb|` would be better if `na!=-nb`, but in the current version of Ferrite I think it always will be. (A similar approach would be required for `getdetJdV`)

*Probably, this should be supported. (Might be technically possible already, with ForwardDiff and some custom initialization of CellValues containing Dual numbers, but probably not very easy nor convenient).

Reply:
Definitely a good point and I agree that detJdV should be different - this raises further confusion regarding what "face integral" means in this context, because the faces do not match anymore.

A bit off-topic, but regarding an Eulerian description we should be able to "just add the displacements to the nodes", so it should not be that difficult. But yes we are missing some convenience functions here.

Reply:
Maybe a compromise is to provide the common definitions with dot and dyadic product along some detailed explanation in the docs (maybe with a guide on on how to implementent custom jumps). What do you think?

Reply:
I would vote for a *single* definition for use in the convenience. 
My suggestion would be the one I suggested [above](https://github.com/Ferrite-FEM/Ferrite.jl/pull/743#discussion_r1257143318) and note that it assumes matching faces. 

As you say, it should be possible to obtain the function values, normals, etc. for each part individually. And that should, as you say, be documented; perhaps even with alternative ways to define the jump if needed!

> A bit off-topic, but regarding an Eulerian description we should be able to "just add the displacements to the nodes", so it should not be that difficult. But yes we are missing some convenience functions here.

Put an issue for discussion/future reminder: https://github.com/Ferrite-FEM/Ferrite.jl/issues/761

Reply:
What should we do about this? Since there are concurring definitions for the jump, maybe we should not provide a default jump in the framework but in the corresponding examples?